core/envelope-sim/src/main/kotlin/fr/sncf/osrd/envelope_sim/etcs/ETCSBrakingCurves.kt

```diff
@@ -233,44 +233,43 @@ private fun computeBrakingCurve(
     val overlayBuilder =
         ConstrainedEnvelopePartBuilder(
             partBuilder,
-            PositionConstraint(0.0, targetPosition),
+            PositionConstraint(0.0, Double.POSITIVE_INFINITY),
             SpeedConstraint(targetSpeed, EnvelopePartConstraintType.FLOOR),
             EnvelopeConstraint(envelope, EnvelopePartConstraintType.CEILING)
         )
-    EnvelopeDeceleration.decelerate(
-        context,
-        targetPosition,
-        targetSpeed,
-        overlayBuilder,
-        -1.0,
-        brakingType
-    )
-    var brakingCurve = partBuilder.build()
-
-    if (brakingType == BrakingType.ETCS_EBD && targetSpeed != 0.0) {
-        // TODO: by doing this, there is an approximation on the gradient used. TBD at a later date.
-        // When target is an LOA, EBD reaches target position at target speed + dVEbi: shift
-        // envelope to make it so. See Subset 026: §3.13.8.3.1, figure 40.
+    if (brakingType != BrakingType.ETCS_EBD || targetSpeed == 0.0) {
```

Comment:
nit: for code-contiguity, I like better the condition to be "positively" related to the case computed.
So I'd either change this condition to:
```suggestion
    if (brakingType == BrakingType.ETCS_SBD || brakingType == BrakingType.ETCS_GUI) {
```
Or just revert the if/else parts (and so the condition)... Or just narrow cases for each computing and reject (for now) EBD+0.0?

Looks like we could "silently" trigger SvL computing if we already iterated on the values (which is not the case).

Reply:
Yeah this wouldn't work. With your condition, we wouldn't go in the if case if it was an svl (EBD + targetSpeed = 0.0). I'll just revert the if else parts.